src/panels/config/zha/zha-cluster-commands.ts

```diff
@@ -0,0 +1,275 @@
+import {
+  html,
+  LitElement,
+  PropertyDeclarations,
+  PropertyValues,
+} from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-card/paper-card";
+import "@polymer/iron-flex-layout/iron-flex-layout-classes";
+import "../../../components/buttons/ha-call-service-button";
+import "../../../components/ha-service-description";
+import "../ha-config-section";
+
+import { HomeAssistant } from "../../../types";
+import "../../../resources/ha-style";
+import { HassEntity } from "home-assistant-js-websocket";
+import {
+  Cluster,
+  Command,
+  ItemSelectedEvent,
+  IssueCommandServiceData,
+  ZHADeviceEntity,
+} from "./types";
+export class ZHAClusterCommands extends LitElement {
+  public hass?: HomeAssistant;
+  public isWide?: boolean;
+  public selectedNode?: HassEntity;
+  public selectedEntity?: ZHADeviceEntity;
+  public selectedCluster?: Cluster;
+  private _showHelp: boolean;
+  private _haStyle?: DocumentFragment;
+  private _ironFlex?: DocumentFragment;
+  private _commands: Command[];
+  private _selectedCommandIndex: number;
+  private _manufacturerCodeOverride?: number;
+  private _issueClusterCommandServiceData?: IssueCommandServiceData;
+
+  constructor() {
+    super();
+    this._showHelp = false;
+    this._selectedCommandIndex = -1;
+    this._commands = [];
+  }
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      isWide: {},
+      selectedNode: {},
+      selectedEntity: {},
+      selectedCluster: {},
+      _showHelp: {},
+      _commands: {},
+      _selectedCommandIndex: {},
+      _manufacturerCodeOverride: {},
+      _issueClusterCommandServiceData: {},
+    };
+  }
+
+  protected update(changedProperties: PropertyValues): void {
+    if (changedProperties.has("selectedCluster")) {
+      this._commands = [];
+      this._selectedCommandIndex = -1;
+      this._fetchCommandsForCluster();
+    }
+    super.update(changedProperties);
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      ${this.renderStyle()}
+      <ha-config-section .isWide="${this.isWide}">
+        <div style="position: relative" slot="header">
+          <span>Cluster Commands</span>
+          <paper-icon-button
+            class="toggle-help-icon"
+            @click="${this._onHelpTap}"
+            icon="hass:help-circle"
+          >
+          </paper-icon-button>
+        </div>
+        <span slot="introduction">View and issue cluster commands.</span>
+
+        <paper-card class="content">
+          ${this._renderCommandPicker()}
```

Comment:
I prefer that all the html not be extracted into multiple functions, but I don't think we have a hard and fast rule on that. Just my personal preference 😄

Reply:
@balloob I copied from the styling from the newer cloud config panels. What is the preference here? I don't have a strong opinion either way...

Reply:
I don't have a strong opinion either. I think that if templates are starting to get long, it might help with readability.